CMakeLists.txt

```diff
@@ -57,6 +57,17 @@ function(add_cxx_flag_if_supported flag)
     endif()
 endfunction()
 
+function(check_if_cxx_compiler_flag_supported flag out_var)
+    if(flag MATCHES "^-Wno-")
+        # Compilers ignore unknown -Wno-foo flags, so look for -Wfoo instead.
```

Comment:
If we're assuming unrecognized -Wno flags will be ignored can't we just set -Wno-unused-but-set-variable unconditionally?

Reply:
I had originally just disabled it for the project, but I got some push back, so I disabled it just for the 3rd party items. I guess it is a valid warning, similar to the `-Wunused-variable` warning that is already enabled. There were only 2 places where this warning was found.

Reply:
If passing -Wno to the compiler doesn't result in an error we don't really care if it's actually doing anything or not, so doing complicated things to try to check if it's actually disabling a warning seems pointless.

Reply:
I see your point and I can change it if you prefer, but I copied this function wholesale from this commit: https://github.com/realm/realm-core/commit/ca39b9fd03864b771cba34471ea1f7076487266c and it also matches the existing `add_cxx_flag_if_supported` function already in the same _CMakeLists.txt_ file.

Reply:
IIRC unrecognized -Wno-blah flags aren't exactly ignored. There are in a weird Schrödinger state where they act as-if ignored unless the compiler prints anything else, in which case it also prints a warning about an unrecognized flag. This can be annoying if you pass tons of -Wno-blah flags unconditionally. Note that we are already doing something like this: https://github.com/realm/realm-core/blob/063927de66f79a0afffbbe36c0bb14d27deba8f2/CMakeLists.txt#L46-L58

I added that logic based on [this code](https://github.com/mongodb/mongo/blob/75228bcd11ef005a55816d76e6c336448a9e1931/SConstruct#L3493-L3503) from the mongo build scripts.

We should probably dedupe this logic between the functions though.